packages/components/src/core/NavigationHeader/components/NavigationHeaderPrimaryNav/index.tsx

```diff
@@ -42,7 +204,9 @@ export default function NavigationHeaderPrimaryNav<T extends string>({
             active={isActive}
             onClick={(e) => {
               onChange(key);
-              item.onClick?.(e);
+              if ("onClick" in item) {
```

Comment:
nit: feels like we could remove this if statement, since we already optional chaining 😄 ?

```ts
onClick={(e) => {
  onChange(key);
  (item as TextNavigationHeaderPrimaryNavItem<T>).onClick?.(e);
}}
```

Reply:
Oh, you’re right, thanks for pointing that out! 😬